test/e2e_node/init_containers_test.go

```diff
@@ -303,43 +553,41 @@ func parseOutput(pod *v1.Pod) containerOutputList {
 	return res
 }
 
-func getContainerOrderingPod(podName string, containerConfigs ...containerTestConfig) *v1.Pod {
-	p := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: podName,
+func preparePod(pod *v1.Pod) {
+	var defaultResourceRequirements v1.ResourceRequirements = v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceMemory: resource.MustParse("15Mi"),
 		},
-		Spec: v1.PodSpec{
-			RestartPolicy: v1.RestartPolicyNever,
+		Limits: v1.ResourceList{
+			v1.ResourceMemory: resource.MustParse("15Mi"),
 		},
 	}
 
-	for _, cc := range containerConfigs {
-		if cc.Name == "" {
-			framework.Failf("expected container config to have a name, found %#v", cc)
-		}
-		c := v1.Container{
-			Name:    cc.Name,
-			Image:   busyboxImage,
-			Command: cc.Command(),
-			Resources: v1.ResourceRequirements{
-				Requests: v1.ResourceList{
-					v1.ResourceMemory: resource.MustParse("15Mi"),
-				},
-				Limits: v1.ResourceList{
-					v1.ResourceMemory: resource.MustParse("15Mi"),
-				},
+	for _, c := range pod.Spec.Containers {
+		c.Resources = defaultResourceRequirements
+		c.VolumeMounts = []v1.VolumeMount{
+			{
+				Name:      "persistent",
+				MountPath: "/persistent",
 			},
 		}
-
-		switch cc.Type {
-		case containerTypeInit:
-			p.Spec.InitContainers = append(p.Spec.InitContainers, c)
-		case containerTypeRegular:
-			p.Spec.Containers = append(p.Spec.Containers, c)
-		default:
-			framework.Failf("expected container config to have a valid type, found %#v", cc)
+	}
+	for _, c := range pod.Spec.InitContainers {
```

Comment:
The volume mounts aren't getting added to the containers, PR https://github.com/SergeyKanzhelev/kubernetes/pull/5 to fix.